borg/platform_windows.pyx

```diff
@@ -0,0 +1,295 @@
+    cdef uint32_t sid_type
+    for i in range(0, length):
+        permissions = None
+        name = ""
+        sidstr = ""
+        if ACEs[i].Trustee.TrusteeForm == TRUSTEE_IS_SID:
+            name, domain, type = _look_up_account_sid(<BYTE*>(ACEs[i].Trustee.ptstrName))
+            sidstr = sid2string(<PSID>(ACEs[i].Trustee.ptstrName))
+
+        elif ACEs[i].Trustee.TrusteeForm == TRUSTEE_IS_NAME:
+            sid_type = 0
+            domainlength = 0
+            LookupAccountNameW(NULL, ACEs[i].Trustee.ptstrName, NULL, &(length), NULL, &domainlength, &sid_type)
+
+            newsid = <PSID>malloc((length) * sizeof(BYTE))
+            domainlength = 0
+            LookupAccountNameW(NULL, ACEs[i].Trustee.ptstrName, newsid, &length, NULL, &domainlength, &sid_type)
+            trusteeName, domain, type = _look_up_account_sid(newsid)
+
+            name = trusteeName
+            sidstr = sid2string(newsid)
+            free(newsid)
+
+        elif ACEs[i].Trustee.TrusteeForm == TRUSTEE_BAD_FORM:
+            continue
+        else:
+            pass
+            # TODO
+        permissions = {'user': {'name': name, 'sid': sidstr}, 'permissions': (ACEs[i].grfAccessPermissions, ACEs[i].grfAccessMode, ACEs[i].grfInheritance)}
+        pyDACL.append(permissions)
+    item[b'dacl'] = safe_encode(json.dumps(pyDACL))
+
+    free(SD)
+    LocalFree(<HLOCAL>ACEs)
+
+def acl_set(path, item, numeric_owner=False):
+    if b'dacl' not in item:
+        return
+
+    pyDACL = json.loads(safe_decode(item[b'dacl']))
+    cdef _EXPLICIT_ACCESS_W* ACEs = <_EXPLICIT_ACCESS_W*>calloc(sizeof(_EXPLICIT_ACCESS_W), len(pyDACL))
+    cdef wchar_t* temp
+    cdef PSID newsid
+    for i in range(0, len(pyDACL)):
+        if pyDACL[i]['user']['name'] == '' or numeric_owner:
+            ACEs[i].Trustee.TrusteeForm = TRUSTEE_IS_SID
+            temp = PyUnicode_AsWideCharString(pyDACL[i]['user']['sid'], NULL)
+            ConvertStringSidToSidW(temp, &newsid)
+            ACEs[i].Trustee.ptstrName = <LPCTSTR>newsid
+            PyMem_Free(temp)
+        else:
+            ACEs[i].Trustee.TrusteeForm = TRUSTEE_IS_NAME
+            ACEs[i].Trustee.ptstrName = PyUnicode_AsWideCharString(pyDACL[i]['user']['name'], NULL)
+        ACEs[i].grfAccessPermissions = pyDACL[i]['permissions'][0]
+        ACEs[i].grfAccessMode = pyDACL[i]['permissions'][1]
+        ACEs[i].grfInheritance = pyDACL[i]['permissions'][2]
+    cdef _ACL* newDACL
+    SetEntriesInAclW(len(pyDACL), ACEs, NULL, &newDACL)
+    cdef wchar_t* cstrPath = PyUnicode_AsWideCharString(path, NULL)
+    SetNamedSecurityInfoW(cstrPath, SE_FILE_OBJECT, DACL_SECURITY_INFORMATION, NULL, NULL, newDACL, NULL)
+
+    for i in range(0, len(pyDACL)):
+        if pyDACL[i]['user']['name'] == '' or numeric_owner:
+            LocalFree(<HLOCAL>ACEs[i].Trustee.ptstrName)
+        else:
+            PyMem_Free(ACEs[i].Trustee.ptstrName)
+    free(ACEs)
+    PyMem_Free(cstrPath)
+    LocalFree(<HLOCAL>newDACL)
```

Comment:
in general: some prints / TODOs left. maybe use a logger?

doesn't such code already exist somewhere? there is a pywin32 package, maybe other 3rd party packages for windows on pypi - did you check them?